src/aosm/azext_aosm/inputs/base_input.py

```diff
@@ -10,19 +10,19 @@ class BaseInput(ABC):
     :type artifact_name: str
     :param artifact_version: The version of the artifact.
     :type artifact_version: str
-    :param default_config: The default configuration for the input. Defaults to None.
-    :type default_config: Optional[Dict[str, Any]]
+    :param default_config_path: The path to the default configuration file for the input. Defaults to None.
+    :type default_config_path: Optional[str]
     """
 
     def __init__(
         self,
         artifact_name: str,
         artifact_version: str,
-        default_config: Optional[Dict[str, Any]] = None,
+        default_config_path: Optional[str] = None,
```

Comment:
If you change this, you need to change all child classes. I'd suggest it would be easier to leave the default_config code (and processing) as it was, and add a `chart_path` or similar parameter to the `HelmChartInput` class.

Reply:
Yes, I missed that, my bad